app/services/articles/destroyer.rb

```diff
@@ -4,8 +4,8 @@ module Destroyer
 
     def call(article, event_dispatcher = Webhook::DispatchEvent)
       article.destroy!
-      Notification.remove_all_without_delay(notifiable_id: article.id, notifiable_type: "Article", action: "Published")
-      Notification.remove_all(notifiable_id: article.id, notifiable_type: "Article", action: "Reaction")
+      Notification.remove_all_without_delay(notifiable_ids: article.id, notifiable_type: "Article")
+      Notification.remove_all(notifiable_ids: article.comments.pluck(:id), notifiable_type: "Comment")
```

Comment:
same thing about guarding with an `if article.comments.exists?` though I think this should be called before the `destroy!` up there because the comments could be gone at this point.

Reply:
We don't have `dependent: :destroy` or `delete_all` on article comments, so they should exist.
I agree that guarding makes sense.